src/Components/Table/TableData.tsx

```diff
@@ -5,35 +5,47 @@ import styles from "../../styles/TableData.module.scss";
 //@ts-ignore
 import { Table, TableHeader, TableRow, TableHead, TableBody, TableCell, IconButton } from "technogetic-iron-smart-ui";
 
-export interface StudentProfile {
-    Course: string;
-    Mobile: string;
-    Student: string;
-    StudentName: string;
-    studentID: string;
+export interface UserProfile {
+    fullName: string;
+    userName: string;
+    email: string;
+    role: string;
+    // emails: string;
+    // leadPlayer: string;
+    // id: string;
 }
+export interface TeamsProfile {
+    emails: string;
+    leadPlayer: string;
+    id: string;
+}
+
+
 
-interface StudentProfilePropsType {
-    studentData: StudentProfile[];
+interface UserProfilePropsType {
+    userData: UserProfile[];
+    teamsData: TeamsProfile[];
     showAdditionalButton?: boolean;
     columns: string[];
 }
-
-interface studentData {
+interface userData {
     [key: string]: string;
 }
 
-const TableData = (props: StudentProfilePropsType) => {
 
-    const [sortedData, setSortedData] = useState(props?.studentData || []);
+const TableData = (props: UserProfilePropsType) => {
+
+    const [sortedData, setSortedData] = useState(props?.userData || []);
     const [isDescending, setIsDescending] = useState(false);
     const [sortKey, setSortKey] = useState("");
 
     useEffect(() => {
-        setSortedData(props?.studentData);
-    }, [props?.studentData]);
+        setSortedData(props?.userData);
```

Comment:
this is wrong is props is undefined then setSortedData will be null which is not the case